Added config and documentation for setup of self-hosted azure (#1391)
conda-forge/conda-forge.github.io: src/maintainer/conda_forge_yml.rst

* :ref:`appveyor`
* :ref:`azure-config`

> * :ref:`self-hosted_azure-config`

Reviewer: I guess this is a leftover. As mentioned earlier we don't need an extra key in conda-forge.yaml to set it up, so let's remove this.
```suggestion
```

      # built packages) as an Azure pipeline artifact that can be downloaded
      store_build_artifacts: False

> .. _self-hosted_azure-config:

Author: Shall i remove this line too?

Reviewer: Good question. Actually I think we should keep it for referring to the config below in the intro.

> Below is the configuration for setting up a self-hosted Azure agent:

Reviewer 2: ```suggestion
Below is an example configuration for setting up a self-hosted Azure agent for linux:
```

> Below is an example configuration for setting up a self-hosted Azure agent for linux:

Reviewer 3: ```suggestion
Below is an example configuration for setting up a self-hosted Azure agent for Linux:
```

      # built packages) as an Azure pipeline artifact that can be downloaded
      store_build_artifacts: False

.. _self-hosted_azure-config:

> self-hosted azure
> -----------------

Reviewer: This does not seem to be the right style used in this file. The sections here correspond to the acceptable top-level keys in `conda-forge.yaml`, but all of the agent settings are under the same key `azure`. I would suggest just to start with a new paragraph, followed by the config sample.
```suggestion
```

> Conda-Smithy is capable of doing things beyond CF's purposes, it also allows setting up
> self-hosted agents for non-CF infrastructures. Here is an example demonstrating that:

Reviewer: A general intro like this should go to [here](https://conda-forge.org/docs/maintainer/infrastructure.html?highlight=smith) as @viniciusdc suggested in https://github.com/conda-forge/conda-forge.github.io/issues/1337#issuecomment-814997213. Let us just focus on what's needed to set it up in this file, something like:
```suggestion
Below is the configuration for setting up a self-hosted Azure agent:
```

Author: >A general intro like this should go to here as @viniciusdc suggested in #1337 (comment). Let us just focus on what's needed to set it up in this file, something like:

Shall  I add all these points under the main "Smithy" heading or make a new heading. (although in my view it will be better to add all these points under the main "Smithy" heading after this line `conda-smithy also contains the command line tool that you should use if you rerender manually from the command line (see Rerendering feedstocks
`
Let me know your views on this, and I will add those points as you suggest

Reviewer: Sounds reasonable to me 🙂

.. code-block:: yaml

>       azure:
>         user_or_org: nsls2forge
>         project_name: nsls2forge
>         for local GPU runners
>         settings_linux:
>         pool:
>         name: BNL-machines
>         demands:
>           - gpu -equals true
>         workspace:
>           clean: all
>         strategy:
>           maxParallel: 2
>           variables:
>             cutensor_path: /home/ci_agent/libcutensor_1.0.1/

Reviewer: Need to fix indentation and clean up some useless keys. This is a better example from the original PR https://github.com/conda-forge/conda-smithy/pull/1296#issue-409225201:
```suggestion
      azure:
        settings_linux:
          pool:
            name: your_local_pool_name
            demands:
              - some_key -equals some_value
          workspace:
            clean: all
          strategy:
            maxParallel: 1
```

> You can also use `Azure virtual machine scale set agents <https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/scale-set-agents?view=azure-devops>`_ 
> for autoscaling agents as they are cheaper to run than permanently active agents.

Reviewer: This should also go to the general intro, not to this place.
```suggestion
```
